_posts/2021-12-04-pdf-protection.adoc

```diff
@@ -0,0 +1,454 @@
+weaknesses (https://en.wikipedia.org/wiki/RC4[Wikipedia]).
+
+Metanorma also defaults to using 256-bit for superior security. But you may find
+legacy PDF readers only supporting 128-bit due to US export restrictions
+previously in place.
+
+In order to generate a encrypted Metanorma PDF output, you will need to provide
+the following document attributes:
+
+* `:pdf-encrypt: true` to specify encryption of PDF output
+
+* `:pdf-encryption-length: {128,256}` to specify encryption strength
+  (128-bit or 256-bit AES keys)
+
+* `:pdf-user-password: {password}` to specify the "user password" needed to
+  open the encrypted PDF document.
+
+[example]
+====
+[source,adoc]
+----
+= My Confidential Document
+:pdf-encrypt: true
+:pdf-encryption-length: 256
+:pdf-user-password: C4nt0p3nW1th0utTh1s
+...
+----
+====
+
+
+== PDF permissions
+
+=== General and the "Owner Password"
+
+PDF permissions are actions normally allowed with unprotected PDFs, but these
+actions can be selectively restricted by the author with a "`owner password`".
+
+If the reader enters the designated "`owner password`", these restrictions can
+be bypassed.
+
+The owner password is set by the following document attribute:
+
+* `:pdf-owner-password: {password}` specifies the PDF "owner password".
+
+[example]
+====
+[source,adoc]
+----
+= Copy Pasta
+:pdf-allow-copy-content: false
+:pdf-owner-password: D0nt700k4tM3
+...
+----
+====
+
+
+=== Disallow copying text or graphics
+
+A PDF file that is marked for copy production means its content is not
+to be copied from the reader view. Commercial publishers typically apply this
+for copyright and content protection concerns (other than DRM).
+
+Sort of as a "`your eyes only`" document, where you can read but not copy out
+content.
+
+This permission can be disallowed using the following document attribute:
+
+* `:pdf-allow-copy-content: false` specifies that copying of content is
+disallowed.
+
+[example]
+====
+[source,adoc]
+----
+= Copy Pasta
+:pdf-allow-copy-content: false
+:pdf-owner-password: D0nt700k4tM3
+...
+----
+====
+//:pdf-allow-edit-content: false
+
+
+=== Disallow editing of content
+
+Editing of PDF content can be disallowed to ensure that the document content remains as when it was originally published.
+
+This permission can be disallowed using the following document attribute:
+
+* `:pdf-allow-assemble-document: false` specifies that editing of PDF content is disallowed.
+
+[example]
+====
+[source,adoc]
+----
+= No Edit
+:pdf-allow-edit-content: false
+:pdf-owner-password: D0nt700k4tM3
+...
+----
+====
+
+=== Disallow inserting, deleting or rotating pages
+
+A PDF file that is marked to disallow page modification means an unauthorized
+user cannot modify its content by inserting, deleting or rotating pages.
+
+This permission can be disallowed using the following document attribute:
+
+* `:pdf-allow-assemble-document: false` specifies that copying of content is
```

Comment:
I don't like this, is "assemble document" an established term? Can we say "reassemble", "rearrange", or "interleave"?

Reply:
From ISO 32000-1:2008: "Assemble" is the correct term.

![image](https://user-images.githubusercontent.com/11865/145185403-0bbf0b8d-2f5e-428f-aeed-61f10c71bbd4.png)